server/create.go

```diff
@@ -221,8 +230,22 @@ func convertFromSafetensors(files map[string]string, baseLayers []*layerGGML, is
 		return nil, err
 	}
 	defer os.RemoveAll(tmpDir)
+	// Set up a root to validate paths
+	root, err := os.OpenRoot(tmpDir)
+	if err != nil {
+		return nil, err
+	}
+	defer root.Close()
 
 	for fp, digest := range files {
+		if !fs.ValidPath(fp) {
+			return nil, fmt.Errorf("%w: %s", errFilePath, fp)
+		}
```

Comment:
Is this not already covered by `root.Stat`?

Reply:
The os.Root doesn't quite work in this case, if you prefix the invalid path with a directory name, than it is seen as safe, since that results in a fs.ErrNotExist

Reply:
Also didn't this already get checked in `CreateHandler?`

Reply:
Yup, checking both since I dont want to rely on this only being called through createHandler, but want to fail fast if I can